src/models/mrt/bgp4mp.rs

```diff
@@ -18,28 +18,33 @@ pub enum BgpState {
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum Bgp4Mp {
-    Bgp4MpStateChange(Bgp4MpStateChange),
-    Bgp4MpStateChangeAs4(Bgp4MpStateChange),
-    Bgp4MpMessage(Bgp4MpMessage),
-    Bgp4MpMessageLocal(Bgp4MpMessage),
-    Bgp4MpMessageAs4(Bgp4MpMessage),
-    Bgp4MpMessageAs4Local(Bgp4MpMessage),
+    StateChange(Bgp4MpStateChange),
+    Message(Bgp4MpMessage),
+}
+
+impl Bgp4Mp {
+    pub const fn msg_type(&self) -> Bgp4MpType {
+        match self {
+            Bgp4Mp::StateChange(x) => x.msg_type,
+            Bgp4Mp::Message(x) => x.msg_type,
+        }
+    }
 }
 
 /// BGP4MP message subtypes.
-#[derive(Debug, Primitive, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Primitive, Copy, Clone, PartialEq, Eq, Hash)]
```

Comment:
Are we keeping the `Primitive` or should we also move the type ids to a separate field? (similar to what you did for attribute types)